DataFormats/TrackerRecHit2D/src/VectorHit.cc

```diff
@@ -80,27 +80,28 @@ bool VectorHit::sharesClusters(VectorHit const& h1, VectorHit const& h2, SharedI
 }
 
 void VectorHit::getKfComponents4D(KfComponentsHolder& holder) const {
-  AlgebraicVector4& pars = holder.params<4>();
+  constexpr int four = 4;
```

Comment:
sorry: somebody shall explain me what is the difference between "four" and "4" in any text, code, or whatever other form of human (and not human) communication

Reply:
My intention was to have a constant with an explanatory name like `nparams` or `ncomponents` or something, which would then enforce consistency throughout the class (not just in one function)

Reply:
Thanks for clarifying, let me get on that...

Reply:
Any reason not to use `theDimension` for that? It gets initialized to 4 in the constructor and can't be changed as far as I can tell.

Reply:
It has to be known at compile time to be used as a template parameter. If `theDimension` is never changed, then it can be made into a constexpr.

Reply:
Since with the suggestion from Erica `theDimension` is now defined as the sum of the dimensions of the 2 VectorHit2D, at least in one of the constructors, I added a new constexpr `nComponents`